pkg/proxy/node.go

```diff
@@ -90,6 +92,79 @@ func (n *NodePodCIDRHandler) OnNodeDelete(node *v1.Node) {
 // OnNodeSynced is a handler for Node syncs.
 func (n *NodePodCIDRHandler) OnNodeSynced() {}
 
+// NodePodCIDRHandler handles the life cycle of kube-proxy based on the node PodCIDR assigned
+// Implements the config.NodeHandler interface
+// https://issues.k8s.io/111321
+type NodeIPsHandler struct {
+	mu      sync.Mutex
+	nodeIPs []net.IP
+	logger  klog.Logger
+}
+
+func NewNodeIPsHandler(ctx context.Context, nodeIPs []net.IP) *NodeIPsHandler {
+	return &NodeIPsHandler{
+		nodeIPs: nodeIPs,
+		logger:  klog.FromContext(ctx),
+	}
+}
+
+var _ config.NodeHandler = &NodeIPsHandler{}
+
+// OnNodeAdd is a handler for Node creates.
+func (n *NodeIPsHandler) OnNodeAdd(node *v1.Node) {
+	n.mu.Lock()
```

Comment:
Why do we need mutex? The informer guarantees that there will be at most 1 handler called at a given time.

Reply:
I see, I must be carrying on with some assumption of legacy code